README.md
# Jake Halloran CS 1520 Project 2
Single page powered by the foundation page for bootstrap with heavy modifications by me.
The Contact form and search relies on ajax and php therefore it was tested using XAMPP for windows
and the newest chrome build for which it works.

Notes:
  Pretty sure the php and CSS should be responsive everywhere, but wouldn't be too surprised if
  something like the contact form ends up running away on window resizing.


changes.txt
Project 2 Changes - Jake Halloran

  For project 1, my website while technically functional and requirements fulfilling, was simply just
ugly, nonresponsive, and sort of pointless. This time around however, most all of that has done a complete
180 degree change. Through extensive use of the bootstrap template freelancer to handle portions of the CSS
I am terrible at I was able to make a site that actually lives up to my expectations of what proper web design
looks like. The inclusion of the ability to auto center elements and handle page resizing much nicer than my
attempts of margin: 0 auto; means that top to bottom the site looks and feels better. Other than just CSS changes,
this new and improved site makes proper use of PHP. For my first site, I technically had some PHP but all it
did was handle a contact form by writing the unmodified JSON object to disk with out even stripping the data
headers. This time around however, both a responsive search bar and a new iteration of my attempt at a contact
form utilize PHP data storage the correct way. Both the search and contact use ajax to post the data to a PHP
function that opens a connection to the database and creates the database and table if it doesn't exist and if
it does merely uses semantic PHP mysqli objects to perform all of the storage and loading operations. As far as
javascript and jquery goes, other than the addition of the necessary js files that happily come with bootstrap,
not much was changed. Frankly, my jquery was by far the most functional part of the site the first time around
and outside of a few changes to the ajax call and removal of functions operating on no longer existing objects,
my jquery remains remarkably intact despite the migration to a single page design. Finally, the sites content
is infinitely better than it was for project part 1. Unlike the first project where I had a few buttons that
showed tiny snippets of content and other pages to show a contact form and resume iFrame, I have now changed
everything to exist on a single page and therefore look much more complete. Additionally, the addition of the
interests search means that much more content about myself is immediately available than on my first site, thus
not only making the content look nicer but also deepening it.


resume.php
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">
    <meta author="Jake Halloran">
    <title>Personal Site: Jake Halloran</title>
    <link rel="stylesheet" href="./css/projectCSS.css">
  </head>

  <body id="contactBody">
    <header id="header">
      <div id="titleNavBox">
        <div id="titleDiv">
          <h1 class="title-text">
            Jake Halloran <br>Programmer and UPITT Student
          </h1>
        </div>

        <div id="navId">
          <nav>
            <ul>
              <li id="navBar">
                <a href="index.php"> Home </a>
                <a href="resume.php"> Resume </a>
                <a href="contact.php"> Contact </a>
                <a href="https://github.com/j-Halloran"> GitHub </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>

    <div id="mainContent">
      <p id="resumeHead">Resume - Last Updated: October 2nd, 2016</p>

      <div id="resumeContainer">
        <iframe id="resumeFrame" src="https://drive.google.com/file/d/0Byj-EsQtcQr8Y19RbUswOHpKb2M/preview" width="1280" height="960" seamless="seamless"></iframe>
      </div>
    </div>

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
    <script src="./js/project2_resume.js"></script>
  </body>
</html>


contact.php
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">
    <meta author="Jake Halloran">
    <title>Personal Site: Jake Halloran</title>
    <link rel="stylesheet" href="./css/projectCSS.css">
  </head>

  <body id="contactBody">
    <header id="header">
      <div id="titleNavBox">
        <div id="titleDiv">
          <h1 class="title-text">
            Jake Halloran <br>Programmer and UPITT Student
          </h1>
        </div>

        <div id="navId">
          <nav>
            <ul>
              <li id="navBar">
                <a href="index.php"> Home </a>
                <a href="resume.php"> Resume </a>
                <a href="contact.php"> Contact </a>
                <a href="https://github.com/j-Halloran"> GitHub </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>

    <div id="formHolder">
      <div id="innerForm">
        <form name="name-form" method="get" action="/" id="commentForm">
            <input id="name" name="name-field" class="textbox-input" type="text" placeholder="First and Last Name" autofocus="autofocus"><br>
            <input id="email" name="email-field" class="textbox-input" type="text" placeholder="Email Address"><br>
            <textarea id="comment" name="comment-field" class="textbox-input" type="text" placeholder="Enter your Comment Here"></textarea><br>
            <input id="submitButton" class="about-button" name="Submit" type="submit">
        </form>
      </div>
    </div>

    <div id="formTitle">
      <p id="titlePara"><br>Enter Your Contact Information</p>
    </div>

    <div id="errorHolder">
      <p id="nameError"><br>You Must Enter A Name</p>

      <p id="emailError"><br>You Must Enter An Email</p>

      <p id="emailFormatError"><br>Your email must contain an @ symbol</p>

      <p id="commentError"><br>You Must Enter A Comment</p>

    </div>

    <div id="successHolder">
      <p id="successPara"><br>Thanks, I will get back to you soon!</p>

    </div>

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
    <script src="./js/project2_form.js"></script>
  </body>
</html>


php/post.php
<?php
  $host = $_SERVER['HTTP_HOST'];
  $table = 'contact_table';
  $servername = 'localhost';
  $username = 'username';
  $password = 'password';
  $dbname = 'contact_db';
  $email ='';
  $name = '';
  $comment ='';

  foreach ($_POST as $k => $v) {
    if($k == "comment-field"){
      $comment = mysql_real_escape_string($v);
    }
    elseif($k=="email-field"){
      $email = mysql_real_escape_string($v);
    }
    elseif($k=="name-field"){
      $name = mysql_real_escape_string($v);
    }
  }

  //Database stuff
  $conn = new mysqli($servername,$username,$password);
  $sql = 'CREATE DATABASE IF NOT EXISTS contact_db';
  if(!$conn->query($sql) === true){
    echo "Error creating database: ". $conn->error;
  }

  //reopen connection now that database is sure to exist
  $conn = new mysqli($servername, $username, $password, $dbname);
  // Check connection
  if (!$conn) {
    die("Connection failed: " . mysqli_connect_error());
  }
  //check for the table, create it if necessary
  if(mysqli_num_rows($conn->query("SHOW TABLES LIKE '".$table."'"))<1){
    $sql = "CREATE TABLE contact_table (
      id INT(6) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
      name VARCHAR(100) NOT NULL,
      email VARCHAR(100) NOT NULL,
      comment VARCHAR(1000) NOT NULL
    )";

    if (!$conn->query($sql) === TRUE) {
      echo "Error creating table: " . $conn->error;
    }
  }

  //store the data in the table
  $sql = "INSERT INTO contact_table (name,email,comment)
  VALUES ('$name','$email','$comment')";
  if (!$conn->query($sql) === TRUE) {
    echo "Error: " . $sql . "<br>" . $conn->error;
  }
?>


php/interest.php
<?php
  $host = $_SERVER['HTTP_HOST'];
  $table = 'FAVORITES_TABLE';
  $servername = 'localhost';
  $username = 'username';
  $password = 'password';
  $dbname = 'favorites';
  $favorite = '';
  $category = '';


  foreach ($_POST as $k => $v) {
    $category = $k;
  }

  //open db connection
  $conn = new mysqli($servername, $username, $password, $dbname);
  // Check connection
  if (!$conn) {
    die("Connection failed: " . mysqli_connect_error());
  }

  $sql = "SELECT category, favorite FROM FAVORITES_TABLE";
  $result = $conn->query($sql);

  if(mysqli_num_rows($result) > 0){
    while($row = $result->fetch_assoc()){
      if($row["category"]==$category){
        $favorite = "Jake's favorite ".$category." is ".$row['favorite']."!";
        break;
      }
    }
  }

  if($favorite == ''){
    $favorite = "Sorry, that category was not found, try album, food, band etc...";
  }

  echo $favorite;
?>


js/project2_form.js
$(document).ready(function(){
  //Hide the error and success messages to start
  $("#errorHolder").hide();
  $("#successHolder").hide();
  $("#formTitle").show();
  $("html, body").animate({ scrollTop: 0 }, "slow");
  //Event listener for submit button
  $("#commentForm").submit(function(event){
    if(verifyForm()){
      $("#formHolder").hide();
      $("#errorHolder").hide();
      $("#formTitle").hide();
      $("#successHolder").show();

      //Use ajax to pass form data without a page refresh
      $.ajax({
            type: 'post',
            url: './php/post.php',
            data: $('#commentForm').serialize(),
            success: function (response) {
              console.log(response);
            }
      });
    }

    else{
      $("#errorHolder").show();
    }

    return false; //dont refresh page
  });
});

function verifyForm(){
  //Hide the titlePara
  $("#formTitle").hide();

  //Hide each individual error
  $("#nameError").hide();
  $("#emailError").hide();
  $("#commentError").hide();
  $("#emailFormatError").hide();

  //check for name length
  if($("input[name='name-field']").val().length==0){
    $("formTitle").hide();
    $("#nameError").show();
    return false;
  }

  //check for email length
  else if($("input[name='email-field']").val().length==0){
    $("formTitle").hide();
    $("#emailError").show();
    return false;
  }

  //check for @ in email
  else if($("input[name='email-field']").val().indexOf('@')==-1){
    $("formTitle").hide();
    $("#emailFormatError").show();
    return false;
  }

  //Check for comment length
  else if($("textarea[name='comment-field']").val().length==0){
    $("formTitle").hide();
    $("#commentError").show();
    return false;
  }
  return true;
}


js/project2_about.js
$(document).ready(function(){
  //Hide the interests and languages sections
  $("#languagesDiv").hide();
  $("#otherInterests").hide();
  $("#classesDiv").hide();
  $("#introDiv").show();

  //hide the search response
  $("#intResponsePara").hide()

  //Event listener for intro button
  $("#intro").click(function(event){
    $("#languagesDiv").hide();
    $("#otherInterests").hide();
    $("#classesDiv").hide();
    $("#introDiv").show();
  });

  //Event listener for languages Button
  $("#languages").click(function(event){
    $("#otherInterests").hide();
    $("#introDiv").hide();
    $("#languagesDiv").show();
    $("#classesDiv").show();
  });

  //Event listener for interests button
  $("#interests").click(function(event){
    $("#introDiv").hide();
    $("#languagesDiv").hide();
    $("#classesDiv").hide();
    $("#otherInterests").show();
    $("#interestsPicture1").hide();
  });

  //Event listener for search
  $("#interestsForm").submit(function(event){
      //Use ajax to pass form data without a page refresh
      $.ajax({
            type: 'post',
            url: './php/interest.php',
            data: $('#searchBar').val(),
            success: function (response) {
              $("#intResponsePara").text(response);
              $("#intResponsePara").show();
              $('#searchBar').val('');
            }
      });

    return false; //dont refresh page
  });
});


index.php
<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Personal Website for Jake Halloran">
    <meta name="author" content="Jake Halloran">

    <title>Jake Halloran - Personal Site</title>

    <!-- Bootstrap Core CSS -->
    <link href="vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">

    <!-- Theme CSS -->
    <link href="css/ProjectCSS.css" rel="stylesheet">

    <!-- Custom Fonts -->
    <link href="vendor/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
    <link href="https://fonts.googleapis.com/css?family=Montserrat:400,700" rel="stylesheet" type="text/css">
    <link href="https://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic" rel="stylesheet" type="text/css">

    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
    <![endif]-->

</head>

<body id="page-top" class="index">

    <!-- Navigation -->
    <nav id="mainNav" class="navbar navbar-default navbar-fixed-top navbar-custom">
        <div class="container">
            <!-- Brand and toggle get grouped for better mobile display -->
            <div class="navbar-header page-scroll">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                    <span class="sr-only">Toggle navigation</span> Menu <i class="fa fa-bars"></i>
                </button>
                <a class="navbar-brand" href="#page-top">Jake Halloran</a>
            </div>

            <!-- Collect the nav links, forms, and other content for toggling -->
            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav navbar-right">
                    <li class="hidden">
                        <a href="#page-top"></a>
                    </li>
                    <li class="page-scroll">
                        <a href="#about">About</a>
                    </li>
                    <li class="page-scroll">
                        <a href="#interests">Interests</a>
                    </li>
                    <li class="page-scroll">
                      <a href="#portfolio">Resume</a>
                    </li>
                    <li class="page-scroll">
                        <a href="#contact">Contact</a>
                    </li>
                    <li class="page-scroll">
                      <a href ="https://www.github.com/j-halloran" target="_blank">GitHub</a>
                    </li>
                </ul>
            </div>
            <!-- /.navbar-collapse -->
        </div>
        <!-- /.container-fluid -->
    </nav>

    <!-- Header -->
    <header>
        <div class="container">
            <div class="row">
                <div class="col-lg-12">
                    <img class="img-responsive" src="img/profile.jpg" id="profPic" alt="">
                    <div class="intro-text">
                        <span class="name">Jake Halloran</span>
                        <hr class="star-light">
                        <span class="skills">Computer Engineer - Accomplished Idiot - Functionally Incompetent Programmer</span>
                    </div>
                </div>
            </div>
        </div>
    </header>

    <!-- About Section -->
    <section class="success" id="about">
        <div class="container">
            <div class="row">
                <div class="col-lg-12 text-center">
                    <h2>About Me</h2>
                    <hr class="star-light">
                </div>
            </div>
            <div class="row">
                <div class="col-lg-4 col-lg-offset-2">
                    <p>  I am Jake Halloran, a computer engineering student at the University of Pittsburgh. I am a senior according to credits but only in my third year of school.</p>
                </div>
                <div class="col-lg-4">
                    <p>Somehow I have managed to cobble together knoweldge in C, C++, Java, Python, Javascript, Model Sim, Autodesk, HTML/CSS, and more.</p>
                </div>
                <div class="col-lg-8 col-lg-offset-2 text-center">
                    <a href="https://www.github.com/j-halloran" class="btn btn-lg btn-outline">
                        <i class="fa fa-download"></i> Visit My GitHub
                    </a>
                </div>
            </div>
        </div>
    </section>

    <!--interests Section-->
    <section class="success2" id="interests">
        <div class="container">
            <div class="row">
                <div class="col-lg-12 text-center">
                    <h2>My Interests</h2>
                    <hr class="star-light">
                </div>
            </div>
            <div class="row">
                <div class="col-lg-4 col-lg-offset-2">
                    <p>Search below to find things that I consider my favorite in a category (ex food or book or album)</p>
                </div>
                <div class="col-lg-4">
                    <p>On top of what is listed below, generally I am down to talk about anything even remotely techy or involving the Steelers or Penguins.</p>
                </div>
                <div class="col-lg-8 col-lg-offset-2 text-center">
                  <div id="interestsFormHolder">
                    <div id="interestsForm">
                      <form name="interest-form" method="get" action="/" id="interestsForm" autocomplete="off">
                          <input type="text" name="search" id="searchBar" placeholder="Search..">
                      </form>
                    </div>
                  </div>
                  <p id="intResponsePara"></p>
                </div>
              </div>
          </div>
    </section>

    <!-- Resume Grid Section -->
    <section id="portfolio" class="success">
        <div class="container">
            <div class="row">
                <div class="col-lg-12 text-center">
                    <h2>Resume</h2>
                    <hr class="star-light">
                </div>
            </div>
            <div class="row">
                <div class="col-lg-12 text-center portfolio-item">
                    <a href="#portfolioModal1" class="portfolio-link" data-toggle="modal">
                        <div class="caption">
                            <div class="caption-content">
                                <i class="fa fa-search-plus fa-3x"></i>
                            </div>
                        </div>
                        <img src="img/portfolio/safe.png" class="img-responsive" alt="my resume">
                    </a>
                </div>
            </div>
        </div>
    </section>


    <!-- Contact Section -->
    <section id="contact">
        <div class="container">
            <div class="row">
                <div class="col-lg-12 text-center">
                    <h2>Contact Me</h2>
                    <hr class="star-primary">
                </div>
            </div>
            <div class="row">
              <div id="formHolder" class="col-lg-12 text-center">
                <div id="innerForm">
                  <form name="name-form" method="get" action="/" id="commentForm">
                      <input id="name" name="name-field" class="textbox-input" type="text" placeholder="First and Last Name" autofocus="autofocus"><br>
                      <input id="email" name="email-field" class="textbox-input" type="text" placeholder="Email Address"><br>
                      <textarea id="comment" name="comment-field" class="textbox-input" type="text" placeholder="Enter your Comment Here"></textarea><br>
                      <input id="submitButton" class="about-button" name="Submit" type="submit">
                  </form>
                </div>
              </div>

              <div id="formTitle">
                <p id="titlePara"><br></p>
              </div>

              <div id="errorHolder">
                <p id="nameError"><br>You Must Enter A Name</p>

                <p id="emailError"><br>You Must Enter An Email</p>

                <p id="emailFormatError"><br>Your email must contain an @ symbol</p>

                <p id="commentError"><br>You Must Enter A Comment</p>

              </div>

              <div id="successHolder">
                <p id="successPara"><br>Thanks, I will get back to you soon!</p>

              </div>
            </div>
        </div>
    </section>

    <!-- Scroll to Top Button (Only visible on small and extra-small screen sizes) -->
    <div class="scroll-top page-scroll hidden-sm hidden-xs hidden-lg hidden-md">
        <a class="btn btn-primary" href="#page-top">
            <i class="fa fa-chevron-up"></i>
        </a>
    </div>


    <!-- Resume Holder -->
    <div class="portfolio-modal modal fade" id="portfolioModal1" tabindex="-1" role="dialog" aria-hidden="true">
        <div class="modal-content">
            <div class="close-modal" data-dismiss="modal">
                <div class="lr">
                    <div class="rl">
                    </div>
                </div>
            </div>
            <div class="container">
                <div class="row">
                    <div class="col-lg-8 col-lg-offset-2">
                        <div class="modal-body">
                            <h2>Resume</h2>
                            <hr class="star-primary">
                        </div>
                    </div>
                </div>
            </div>
            <div id="resumeContainer">
              <iframe id="resumeFrame" src="https://drive.google.com/file/d/0Byj-EsQtcQr8Y19RbUswOHpKb2M/preview" width="1280" height="960" seamless="seamless"></iframe>
            </div>
        </div>
    </div>

    <!-- jQuery -->
    <script src="vendor/jquery/jquery.min.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="vendor/bootstrap/js/bootstrap.min.js"></script>

    <!-- Plugin JavaScript -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-easing/1.3/jquery.easing.min.js"></script>

    <!-- Contact Form JavaScript -->
    <script src="js/project2_form.js"></script>

    <!-- Interests Script-->
    <script src="js/project2_about.js"></script>

    <!-- Theme JavaScript -->
    <script src="js/freelancer.min.js"></script>

</body>

</html>
